core/trino-main/src/main/java/io/trino/node/RemoteNodeState.java

```diff
@@ -185,26 +185,13 @@ else if (!serverInfo.nodeVersion().equals(expectedNodeVersion)) {
                     logWarning("Error fetching node state from %s: %s", infoUri, t.getMessage());
                     return false;
                 }, directExecutor());
+
         future.compareAndSet(null, catching);
-        Futures.addCallback(
-                catching,
-                new FutureCallback<>()
-                {
-                    @Override
-                    public void onSuccess(Boolean result)
-                    {
-                        future.compareAndSet(catching, null);
-                        lastUpdateNanos.set(ticker.read());
-                    }
+        addSuccessCallback(catching, _ -> {
```

Comment:
This misses the failure case?

Reply:
The code above converts the future into a version that can't fail.  I could have made a mistake on that, so please check.